framework/src/modules/chain/submodules/loader.js

```diff
@@ -184,70 +185,54 @@ __private.syncTimer = function() {
 };
 
 /**
- * Gets a random peer and loads signatures from network.
- * Processes each signature from peer.
+ * Loads signatures from network.
+ * Processes each signature from the network.
  *
  * @private
  * @param {function} cb
  * @returns {setImmediateCallback} cb, err
  * @todo Add description for the params
  */
-__private.loadSignatures = function(cb) {
-	async.waterfall(
-		[
-			function(waterCb) {
-				self.getRandomPeerFromNetwork((getRandomPeerFromNetworkErr, peer) => {
-					if (getRandomPeerFromNetworkErr) {
-						return setImmediate(waterCb, getRandomPeerFromNetworkErr);
-					}
-					return setImmediate(waterCb, null, peer);
-				});
-			},
-			function(peer, waterCb) {
-				library.logger.info(`Loading signatures from: ${peer.string}`);
-				peer.rpc.getSignatures((err, res) => {
-					if (err) {
-						modules.peers.remove(peer);
-						return setImmediate(waterCb, err);
-					}
-					return library.schema.validate(
-						res,
-						definitions.WSSignaturesResponse,
-						validateErr => setImmediate(waterCb, validateErr, res.signatures)
-					);
+__private.loadSignatures = async function() {
+	library.logger.info('Loading signatures from the network');
+
+	// TODO: Add target module to procedure name. E.g. chain:getSignatures
+	const result = await library.channel.invoke('network:request', {
+		procedure: 'getSignatures',
+	});
+
+	const validate = promisify(library.schema.validate.bind(library.schema));
+
+	await validate(result, definitions.WSSignaturesResponse);
```

Comment:
are we not handling error if the validation throws some error here?

Reply:
No because the function is `async` so the caolan `async.retry` module will capture the error.